action/action.go

```diff
@@ -40,12 +40,16 @@ type hasDestination interface {
 }
 
 // Sign signs the action using sender's private key
-func Sign(act Envelope, sk crypto.PrivateKey) (SealedEnvelope, error) {
-	sealed := SealedEnvelope{Envelope: act}
-
-	sealed.srcPubkey = sk.PublicKey()
+func Sign(elp Envelope, sk crypto.PrivateKey) (SealedEnvelope, error) {
+	if elp.IsExternal() {
+		panic("not allowed to sign external tx")
```

Comment:
wonder if it is better to return error instead of panic? if it is possible for someone forms a bad request and make the node panic?

Reply:
Sign() only get calls by ioctl, so node won't panic

since RLP-tx would ONLY come from proxy, should never be generated from core or ioctl, it's trying to prevent such mis-use. Return an error might be ignored, panic make it clear that sth is seriously wrong